File: spark/left-anti-join/irrelevant-users/irrelevant_users.scala
import org.apache.spark.sql.functions._

pupil.join(innovation_craft,
           pupil.col("id") === innovation_craft.col("owner_id"),
           "left_anti")
     .select("id", "first_name", "last_name", "grade")
     .orderBy("grade", "id")
     .show()

File: spark/left-anti-join/irrelevant-users/irrelevant_users.py
from pyspark.sql.functions import *

pupil.join(innovation_craft,
           pupil["id"] == innovation_craft["owner_id"],
           "left_anti") \
     .select("id", "first_name", "last_name", "grade") \
     .orderBy("grade", "id") \
     .show()